From b825fa5cb90423f644d192267b3211d8acb71eca Mon Sep 17 00:00:00 2001
From: Giuseppe Maggio <jertlokk@gmail.com>
Date: Tue, 1 Jan 2019 17:26:22 +0000
Subject: [PATCH] angler: QCamera: avoid virtual function override

device/huawei/angler/camera/QCamera2/HAL3/QCamera3Channel.h:256:21:
error: 'qcamera::QCamera3RegularChannel::request' hides overloaded
virtual function [-Werror,-Woverloaded-virtual]
    virtual int32_t request(buffer_handle_t *buffer, uint32_t
frameNumber);
                    ^
device/huawei/angler/camera/QCamera2/HAL3/QCamera3Channel.h:164:21:
note: hidden overloaded virtual function
'qcamera::QCamera3ProcessingChannel::request' declared here: different
number of parameters (4 vs 2)
    virtual int32_t request(buffer_handle_t *buffer,
                    ^
device/huawei/angler/camera/QCamera2/HAL3/QCamera3Channel.h:382:21:
error: 'qcamera::QCamera3YUVChannel::request' hides overloaded virtual
function [-Werror,-Woverloaded-virtual]
    virtual int32_t request(buffer_handle_t *buffer,
                    ^
device/huawei/angler/camera/QCamera2/HAL3/QCamera3Channel.h:164:21:
note: hidden overloaded virtual function
'qcamera::QCamera3ProcessingChannel::request' declared here: different
number of parameters (4 vs 5)
    virtual int32_t request(buffer_handle_t *buffer,
                    ^

Change-Id: I917f5ad1e1c3e622d2a91d8f829f9780dccde843
---
 camera/QCamera2/HAL3/QCamera3Channel.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/camera/QCamera2/HAL3/QCamera3Channel.h b/camera/QCamera2/HAL3/QCamera3Channel.h
index b1edece..9d3a31c 100644
--- a/camera/QCamera2/HAL3/QCamera3Channel.h
+++ b/camera/QCamera2/HAL3/QCamera3Channel.h
@@ -246,7 +246,7 @@ class QCamera3RegularChannel : public QCamera3ProcessingChannel
                     uint32_t postprocess_mask,
                     QCamera3Channel *metadataChannel,
                     uint32_t numBuffers = MAX_INFLIGHT_REQUESTS);
-
+    using QCamera3ProcessingChannel::request;
     virtual ~QCamera3RegularChannel();
 
     virtual int32_t setBatchSize(uint32_t batchSize);
@@ -377,6 +377,7 @@ class QCamera3YUVChannel : public QCamera3ProcessingChannel
             cam_stream_type_t stream_type,
             uint32_t postprocess_mask,
             QCamera3Channel *metadataChannel);
+    using QCamera3ProcessingChannel::request;
     ~QCamera3YUVChannel();
     virtual int32_t initialize(cam_is_type_t isType);
     virtual int32_t request(buffer_handle_t *buffer,
